### PR TITLE
HttpHeaders.ContentMd5 is changed to Contenthash

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

closes https://github.com/Azure/azure-sdk-for-cpp/issues/1213

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp [hunk 1]
```diff
@@ -3734,11 +3735,11 @@ namespace Azure { namespace Storage { namespace Blobs {
               = BlobLeaseStatusFromString(httpResponse.GetHeaders().at("x-ms-lease-status"));
           response.LeaseState
               = BlobLeaseStateFromString(httpResponse.GetHeaders().at("x-ms-lease-state"));
-          auto response_lease_duration_iterator
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:3737]: nit: is this change relevant to the PR?

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp [hunk 2]
```diff
@@ -728,10 +634,44 @@ namespace Azure { namespace Storage { namespace Blobs {
       std::string ETag;
       std::string LastModified;
       std::string CopyId;
-      Blobs::Models::CopyStatus CopyStatus = Blobs::Models::CopyStatus::Unknown;
+      Models::CopyStatus CopyStatus = Models::CopyStatus::Unknown;
       Azure::Core::Nullable<std::string> VersionId;
     }; // struct StartCopyPageBlobIncrementalResult
 
+    struct AppendBlockFromUriResult
```
Comment (katmsft) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:641]: Why the sequence of the model types are changed?

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:3737]: Yep, it's to fix a build error caused by this PR. So it's definitely relevant to this PR.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:641]: They're not changed, they're reordered. This is a defect of this auto-generator.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:3737]: How come a change in this PR affects absolutely unrelated code?
